### DISCUSSION TITLE
Improve your contribution experience and guidelines

### DISCUSSION THREAD

**Discussion by YannickRe** [microsoft/powerplatform-build-tools#525] [General]
Hi team,

When I offered my idea #517, I get the response that it's an Open Source project and I was free to contribute. I have a rather decent track record contributing to various open source projects, but this one could do with some more guidance to help people do what needs to be done. I'm sure if some of below points could be improved on, you can even get complete tasks submitted by community members.

So, here is my input:

1. Open source the @microsoft/powerplatform-cli-wrapper without that, nothing can be tested by mere mortals outside of Microsoft
2. Expand on your Contributing.md file
    - Explain how to set up a local dev environment
    - Explain which type of change goes where (PAC CLI, Wrapper, or this repository)
    - Explain how to locally test DevOps tasks
    - Explain how to submit the PR
3. Remove the requirement to have a branch in this repository to be able to run the tests. This is stupid for contributors, because we don't understand what is happening, it's extremely stupid for maintainers as you have to locally clone which is just double the work.

Probably some more things, but this is already enough work to get done 😅
And please remember that I love this repository, so all this comes from a place of love.

**Comment by ricklarose16** [comment]
Hi,

Can you (Microsoft) please respond to YannickRe's conversation?

The most important to me is point 1. Open source the @microsoft/powerplatform-cli-wrapper without that, nothing can be tested by mere mortals outside of Microsoft...

Thanks!
